### ISSUE TITLE
I don't want to paginate to generate PDF

### ISSUE THREAD

**Issue by 54huige** [wkhtmltopdf/wkhtmltopdf#4853]
I don't want to paginate to generate PDF. Although it's long, the CSS settings don't work.

I don't know the height of the page, I just expect it to generate a PDF page

Is there a good way to do it now? I need it. thank you

**Comment by josemauromani** [comment]
Hi, can you post your code, so we can analyze it. 
Thanks

**Comment by josemauromani** [comment]
Hi, can you post your code, so we can analyze it.
Thanks

**Comment by PhilterPaper** [comment]
Are you asking for an unlimited height PDF (single page), such as for a roll printer? This has been discussed many times -- please research the tickets. There may be practical limits for standards-conforming readers (14400 points, or larger with User Units). So far, I haven't seen a practical way to generate a long PDF. The best so far has been to take a swag at the height, output a single page with that custom media size, and use some utility to trim off the unused bottom of the PDF.

**Comment by 54huige** [comment]
> Are you asking for an unlimited height PDF (single page), such as for a roll printer? This has been discussed many times -- please research the tickets. There may be practical limits for standards-conforming readers (14400 points, or larger with User Units). So far, I haven't seen a practical way to generate a long PDF. The best so far has been to take a swag at the height, output a single page with that custom media size, and use some utility to trim off the unused bottom of the PDF.

Yes, thank you for your reply. I thought about it. How to remove the redundant blank area? Is there a ready-made tool

**Comment by PhilterPaper** [comment]
I seem to recall that there was some discussion on how you could remove the unwanted blank area at the bottom. It may involve reading in the PDF to a utility, and changing the MediaBox, or adding a CropBox. The problem is that the PDF has  0,0 at the lower left corner, so all y-coordinates would have to be adjusted too. Adding a CropBox, if there isn't already one to update, might do the job, but at the cost of rewriting the trailer and cross reference table. Googling "PDF trim" I see that there are tools out there to trim the margins of a page, so it's been done already, but they may be strictly graphical rather than command line.
